packages/firebase_storage/README.md

```diff
@@ -4,40 +4,42 @@
 
 A Flutter plugin to use the [Firebase Cloud Storage API](https://firebase.google.com/products/storage/).
 
-For Flutter plugins for other Firebase products, see [README.md](https://github.com/FirebaseExtended/flutterfire/blob/master/README.md).
+To learn more about Firebase, please visit the [Firebase website](https://firebase.google.com)
 
-## Usage
+## Getting Started
 
-To use this plugin, add `firebase_storage` as a [dependency in your pubspec.yaml file](https://flutter.io/platform-plugins/).
+> If you are migrating your existing project to these new plugins, please start with the [migration guide](https://firebase.flutter.dev/docs/migration)
 
-### Logging
+To get started with Cloud Storage for FlutterFire, please [see the documentation](https://firebase.flutter.dev/docs/storage/overview)
+available at [https://firebase.flutter.dev](https://firebase.flutter.dev/docs/overview)
+
+## Usage
 
-If you wish to see status events for your upload tasks in your logs, you should listen to the `StorageUploadTask.events` stream.  
-This could look like the following if you are using `StorageReference.putData`:
+To use any of the Firebase services, FlutterFire needs to be initialized.  To initialize FlutterFire,
+call the `initializeApp` method on the `Firebase` class:
 
 ```dart
-final StorageReference storageReference = FirebaseStorage().ref().child(path);
+await Firebase.initializeApp();
+```
 
-final StorageUploadTask uploadTask = storageReference.putData(data);
+### Basic Example
 
-final StreamSubscription<StorageTaskEvent> streamSubscription = uploadTask.events.listen((event) {
-  // You can use this to notify yourself or your user in any kind of way.
-  // For example: you could use the uploadTask.events stream in a StreamBuilder instead
-  // to show your user what the current status is. In that case, you would not need to cancel any
-  // subscription as StreamBuilder handles this automatically.
+To upload a file to Firebase Storage, you need to first create a reference for the file, and then call either the
+`putFile` or `putBlob` method that takes the passed data and uploads them to Firebase Storage:
 
-  // Here, every StorageTaskEvent concerning the upload is printed to the logs.
-  print('EVENT ${event.type}');
-});
+```dart
+Future<void> uploadFile() {
+  // Create bucket storage reference to not yet existing file
+  Reference ref = FirebaseStorage.instance.ref('/file-upload-test.txt');
 
-// Cancel your subscription when done.
-await uploadTask.onComplete;
-streamSubscription.cancel();
+  // Upload file
+  UploadTask task = ref.putFile(file);
```

Comment:
Lets add in an example of task usage - listening for events and printing the progress.

Reply:
[Added](https://github.com/FirebaseExtended/flutterfire/pull/3419/commits/43db20f1f4f3ab232a6f5ae76411d80eb6ddf96e)